Phase 1 - Start to try to standardize README.md files.

sgilda opened the pull request:
More to come later

sgilda committed:
Attempt to standardize README.md files.

pmuir commented:
Sande, would it be best to hold it until all is done? Otherwise I can see a loop forming!

On 26 Mar 2012, at 18:50, sgilda wrote:

> More to come later
> 
> You can merge this Pull Request by running:
> 
>  git pull https://github.com/sgilda/quickstart readme-cleanup
> 
> Or you can view, comment on it, or merge it online at:
> 
>  https://github.com/jbossas/quickstart/pull/167
> 
> -- Commit Summary --
> - Phase 1 - Start to try to standardize README.md files.
> 
> -- File Changes --
> 
> M README.md (64)
> M bean-validation/README.md (29)
> M bmt/README.md (98)
> M cdi-injection/README.md (46)
> M cmt/README.md (123)
> M ejb-in-ear/README.md (27)
> M ejb-security/README.md (58)
> M servlet-async/README.md (55)
> M servlet-filterlistener/README.md (52)
> M servlet-security/README.md (63)
> M template/README.md (36)
> 
> -- Patch Links --
> 
>  https://github.com/jbossas/quickstart/pull/167.patch
>  https://github.com/jbossas/quickstart/pull/167.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167

sgilda commented:
That works for me. I thought it would be a while, but I'm trying to get 
them over with now. :-)

On 03/27/2012 12:31 PM, Pete Muir wrote:

> Sande, would it be best to hold it until all is done? Otherwise I can see a loop forming!
> 
> On 26 Mar 2012, at 18:50, sgilda wrote:
> 
> > More to come later
> > 
> > You can merge this Pull Request by running:
> > 
> >   git pull https://github.com/sgilda/quickstart readme-cleanup
> > 
> > Or you can view, comment on it, or merge it online at:
> > 
> >   https://github.com/jbossas/quickstart/pull/167
> > 
> > -- Commit Summary --
> > - Phase 1 - Start to try to standardize README.md files.
> > 
> > -- File Changes --
> > 
> > M README.md (64)
> > M bean-validation/README.md (29)
> > M bmt/README.md (98)
> > M cdi-injection/README.md (46)
> > M cmt/README.md (123)
> > M ejb-in-ear/README.md (27)
> > M ejb-security/README.md (58)
> > M servlet-async/README.md (55)
> > M servlet-filterlistener/README.md (52)
> > M servlet-security/README.md (63)
> > M template/README.md (36)
> > 
> > -- Patch Links --
> > 
> >   https://github.com/jbossas/quickstart/pull/167.patch
> >   https://github.com/jbossas/quickstart/pull/167.diff
> > 
> > ---
> > 
> > Reply to this email directly or view it on GitHub:
> > 
> > ## https://github.com/jbossas/quickstart/pull/167
> > 
> > Reply to this email directly or view it on GitHub:
> > https://github.com/jbossas/quickstart/pull/167#issuecomment-4720541

sgilda commented:
Pete, all I have left are the ws\* readmes at this point.  I'll go through what I have in tonight and make sure it's ok.

sgilda commented:
I will have a few more changes, but it will be a while before I can get to them. Could we go ahead and merge these for now?

sgilda commented:
After review, of course. :-)
Still wondering if I should copy sections instead of link to them. But copy is harder to maintain if things change or the files are edited.

sgilda commented:
You can hold off. I'll try to finish them up today.

sgilda commented:
Finished and squashed the commits. I didn't touch the html5 quickstarts since this isn't the source of record.
Pete, I think they are ready for review.

pmuir commented on README.md:
```diff
@@ -201,26 +201,41 @@ The README for each quickstart will specify which configuration is required to r
 
+1. The command used to build the quickstart depends on the individual quickstart, the server version, and how you configured Maven.
+    * If you are running JBoss AS 7, it uses community artifacts available in the Maven central repository, so command line arguments are not usually required. 
+    * If you are running JBoss Enterprise Application Platform 6 and did not configure the Maven user settings described in [Configure Maven for JBoss Enterprise Application Platform 6](#eap6mavenconfig) above, you need to specify command line arguments. 
```
AS comes after EAP :-)

pmuir commented on README.md:
```diff
@@ -201,26 +201,41 @@ The README for each quickstart will specify which configuration is required to r
+2. The command also depends on whether you are deploying an EAR or a WAR.
+    * For a WAR, you can build and deploy the archive as one step or as two separate steps. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) and [Build and Deploy the Quickstart EAR or WAR in Two Steps](#buildanddeploy2steps) below for details.
+    * For an EAR, the build and deploy must be done in one step. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) below for details.
+3. If you want to redeploy a quickstart that uses a database, you should undeploy it first. This ensures the database is cleaned up correctly before the redeploy.
```
I don't think this is true. A redeploy is (should be!) an alias for undeploy and deploy.

sgilda replied:
In the cmt quickstart, Tom said: If you need to redeploy the war then ensure that you explicitly undeploy it first so that databases are cleaned up correctly.

I wonder if he had issues?

pmuir replied:
Could you check with him?

On 28 Mar 2012, at 17:56, sgilda wrote:

> > @@ -201,26 +201,41 @@ The README for each quickstart will specify which configuration is required to r
> > 
> > Review the README file in the root folder of the quickstart for specific details on how to build and run the example. In most cases you do the following:
> > 
> > +1. The command used to build the quickstart depends on the individual quickstart, the server version, and how you configured Maven.
> > -    \* If you are running JBoss AS 7, it uses community artifacts available in the Maven central repository, so command line arguments are not usually required. 
> > -    \* If you are running JBoss Enterprise Application Platform 6 and did not configure the Maven user settings described in [Configure Maven for JBoss Enterprise Application Platform 6](#eap6mavenconfig) above, you need to specify command line arguments. 
> >   +2. The command also depends on whether you are deploying an EAR or a WAR.
> > -    \* For a WAR, you can build and deploy the archive as one step or as two separate steps. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) and [Build and Deploy the Quickstart EAR or WAR in Two Steps](#buildanddeploy2steps) below for details.
> > -    \* For an EAR, the build and deploy must be done in one step. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) below for details.
> >   +3. If you want to redeploy a quickstart that uses a database, you should undeploy it first. This ensures the database is cleaned up correctly before the redeploy.
> 
> In the cmt quickstart, Tom said: If you need to redeploy the war then ensure that you explicitly undeploy it first so that databases are cleaned up correctly.
> 
> I wonder if he had issues?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167/files#r616183

pmuir commented on README.md:
```diff
@@ -201,26 +201,41 @@ The README for each quickstart will specify which configuration is required to r
+    * If you are running JBoss AS 7, it uses community artifacts available in the Maven central repository, so command line arguments are not usually required. 
+    * If you are running JBoss Enterprise Application Platform 6 and did not configure the Maven user settings described in [Configure Maven for JBoss Enterprise Application Platform 6](#eap6mavenconfig) above, you need to specify command line arguments. 
+2. The command also depends on whether you are deploying an EAR or a WAR.
+    * For a WAR, you can build and deploy the archive as one step or as two separate steps. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) and [Build and Deploy the Quickstart EAR or WAR in Two Steps](#buildanddeploy2steps) below for details.
```
I think we should just drop the two step war deploy. It seems to only add confusion to me.

pmuir commented on README.md:
```diff
@@ -201,26 +201,41 @@ The README for each quickstart will specify which configuration is required to r
+    * For an EAR, the build and deploy must be done in one step. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) below for details.
+3. If you want to redeploy a quickstart that uses a database, you should undeploy it first. This ensures the database is cleaned up correctly before the redeploy.
+4. See the README file in each individual quickstart folder for specific details and information on how to run and access the example.
+5. You can also start the JBoss Application Server and deploy the quickstarts using Eclipse. See the [Getting Started Developing Applications Guide](https://docs.jboss.org/author/display/AS71/Getting+Started+Developing+Applications+Guide "Getting Started Developing Applications Guide") for more information.
```
I think this needs to mention EAP as well.

sgilda replied:
I don't have a link to the Getting Started there yet. Do we even know what the general link will be yet?

pmuir replied:
red.ht/jbeap-6-docs :-)

On 28 Mar 2012, at 18:40, sgilda wrote:

> > @@ -201,26 +201,41 @@ The README for each quickstart will specify which configuration is required to r
> > 
> > Review the README file in the root folder of the quickstart for specific details on how to build and run the example. In most cases you do the following:
> > 
> > +1. The command used to build the quickstart depends on the individual quickstart, the server version, and how you configured Maven.
> > -    \* If you are running JBoss AS 7, it uses community artifacts available in the Maven central repository, so command line arguments are not usually required. 
> > -    \* If you are running JBoss Enterprise Application Platform 6 and did not configure the Maven user settings described in [Configure Maven for JBoss Enterprise Application Platform 6](#eap6mavenconfig) above, you need to specify command line arguments. 
> >   +2. The command also depends on whether you are deploying an EAR or a WAR.
> > -    \* For a WAR, you can build and deploy the archive as one step or as two separate steps. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) and [Build and Deploy the Quickstart EAR or WAR in Two Steps](#buildanddeploy2steps) below for details.
> > -    \* For an EAR, the build and deploy must be done in one step. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) below for details.
> >   +3. If you want to redeploy a quickstart that uses a database, you should undeploy it first. This ensures the database is cleaned up correctly before the redeploy.
> >   +4. See the README file in each individual quickstart folder for specific details and information on how to run and access the example.
> >   +5. You can also start the JBoss Application Server and deploy the quickstarts using Eclipse. See the [Getting Started Developing Applications Guide](https://docs.jboss.org/author/display/AS71/Getting+Started+Developing+Applications+Guide) for more information.
> 
> I don't have a link to the Getting Started there yet. Do we even know what the general link will be yet?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167/files#r616471

sgilda replied:
But of course. I send the link out all the time for review. ;-)

On 03/29/2012 07:21 AM, Pete Muir wrote:

> > @@ -201,26 +201,41 @@ The README for each quickstart will specify which configuration is required to r
> > 
> >   Review the README file in the root folder of the quickstart for specific details on how to build and run the example. In most cases you do the following:
> > 
> > +1. The command used to build the quickstart depends on the individual quickstart, the server version, and how you configured Maven.
> > -    \* If you are running JBoss AS 7, it uses community artifacts available in the Maven central repository, so command line arguments are not usually required.
> > -    \* If you are running JBoss Enterprise Application Platform 6 and did not configure the Maven user settings described in [Configure Maven for JBoss Enterprise Application Platform 6](#eap6mavenconfig) above, you need to specify command line arguments.
> >   +2. The command also depends on whether you are deploying an EAR or a WAR.
> > -    \* For a WAR, you can build and deploy the archive as one step or as two separate steps. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) and [Build and Deploy the Quickstart EAR or WAR in Two Steps](#buildanddeploy2steps) below for details.
> > -    \* For an EAR, the build and deploy must be done in one step. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) below for details.
> >   +3. If you want to redeploy a quickstart that uses a database, you should undeploy it first. This ensures the database is cleaned up correctly before the redeploy.
> >   +4. See the README file in each individual quickstart folder for specific details and information on how to run and access the example.
> >   +5. You can also start the JBoss Application Server and deploy the quickstarts using Eclipse. See the [Getting Started Developing Applications Guide](https://docs.jboss.org/author/display/AS71/Getting+Started+Developing+Applications+Guide) for more information.
> >   red.ht/jbeap-6-docs :-)
> 
> On 28 Mar 2012, at 18:40, sgilda wrote:
> 
> > > @@ -201,26 +201,41 @@ The README for each quickstart will specify which configuration is required to r
> > > 
> > > Review the README file in the root folder of the quickstart for specific details on how to build and run the example. In most cases you do the following:
> > > 
> > > +1. The command used to build the quickstart depends on the individual quickstart, the server version, and how you configured Maven.
> > > -    \* If you are running JBoss AS 7, it uses community artifacts available in the Maven central repository, so command line arguments are not usually required.
> > > -    \* If you are running JBoss Enterprise Application Platform 6 and did not configure the Maven user settings described in [Configure Maven for JBoss Enterprise Application Platform 6](#eap6mavenconfig) above, you need to specify command line arguments.
> > >   +2. The command also depends on whether you are deploying an EAR or a WAR.
> > > -    \* For a WAR, you can build and deploy the archive as one step or as two separate steps. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) and [Build and Deploy the Quickstart EAR or WAR in Two Steps](#buildanddeploy2steps) below for details.
> > > -    \* For an EAR, the build and deploy must be done in one step. See [Build and Deploy the Quickstart EAR or WAR in One Step](#buildanddeploy1step) below for details.
> > >   +3. If you want to redeploy a quickstart that uses a database, you should undeploy it first. This ensures the database is cleaned up correctly before the redeploy.
> > >   +4. See the README file in each individual quickstart folder for specific details and information on how to run and access the example.
> > >   +5. You can also start the JBoss Application Server and deploy the quickstarts using Eclipse. See the [Getting Started Developing Applications Guide](https://docs.jboss.org/author/display/AS71/Getting+Started+Developing+Applications+Guide) for more information.
> > >   I don't have a link to the Getting Started there yet. Do we even know what the general link will be yet?
> > 
> > ---
> > 
> > Reply to this email directly or view it on GitHub:
> > 
> > ## https://github.com/jbossas/quickstart/pull/167/files#r616471
> > 
> > Reply to this email directly or view it on GitHub:
> > https://github.com/jbossas/quickstart/pull/167/files#r620521

pmuir commented on README.md:
```diff
@@ -201,26 +201,41 @@ The README for each quickstart will specify which configuration is required to r
-      * Although some of the quickstarts require special commands, for most of the quickstarts you do the one of the following. 
-         
-          For JBoss Enterprise Application Platform 6, Maven user settings NOT configured: 
+2. Open a command line and navigate to the root of the directory of the quickstart you want to run.
```
Don't all quickstarts require JBoss AS running?

sgilda replied:
Not if you are running a "managed" Arquillian test.

pmuir commented on README.md:
```diff
@@ -201,26 +201,41 @@ The README for each quickstart will specify which configuration is required to r
             mvn clean package -s PATH_TO_QUICKSTARTS/example-settings.xml
-
-          For JBoss AS 7 or JBoss Enterprise Application Platform 6, Maven user settings configured: 
+    * For JBoss AS 7 or JBoss Enterprise Application Platform 6, Maven user settings configured: 
```
The punctuation aren't right here, it reads as "JBoss AS 7, Maven user settings not configured, or JBoss Enterprise Application Platform, Maven uer settings not configured". Maybe use parentheses?

sgilda replied:
Is this OK:  For JBoss AS 7 or JBoss Enterprise Application Platform 6 (Maven user settings configured)

pmuir replied:
Yes, much better.

On 28 Mar 2012, at 18:41, sgilda wrote:

> > ```
> >         mvn clean package -s PATH_TO_QUICKSTARTS/example-settings.xml
> > ```
> > 
> > ## 
> > -          For JBoss AS 7 or JBoss Enterprise Application Platform 6, Maven user settings configured: 
> > -    \* For JBoss AS 7 or JBoss Enterprise Application Platform 6, Maven user settings configured: 
> 
> Is this OK:  For JBoss AS 7 or JBoss Enterprise Application Platform 6 (Maven user settings configured)
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167/files#r616476

pmuir commented on README.md:
```diff
@@ -201,26 +201,41 @@ The README for each quickstart will specify which configuration is required to r
-   4. Alternatively, you can build and deploy the quickstart in one step. As in step 2. above when you built the archive, the command you use depends on the individual quickstart, the server version, and how you configured Maven. Although some of the quickstarts require special commands, for most of the quickstarts you do one of the following:
+
+<a id="buildanddeploy1step"></a>
+#### Build and Deploy the Quickstart EAR or WAR in One Step
```
Same comments as above.

pmuir commented on README.md:
```diff
@@ -283,6 +301,10 @@ You can run these tests using either a managed or remote container.
+<a id="useeclipse"/>
+Use Eclipse to Start the Server and Build and Deploy the Quickstarts
+-------------------------------------
+You can also deploy the quickstarts from Eclipse using JBoss tools. For more information on how to set up Maven and the JBoss tools, refer to the [Getting Started Developing Applications Guide](https://docs.jboss.org/author/display/AS71/Getting+Started+Developing+Applications+Guide "Getting Started Developing Applications Guide").
```
I think we need to rework this to say JBDS e.g.

Use JBoss Developer Studio or Eclipse to Start...

sgilda replied:
This is what I love about moving the instructions here. Only one place to make the change. :-)

sgilda replied:
Oops. The link text will all have to change too. :-(

pmuir commented on bean-validation/README.md:
```diff
@@ -5,37 +5,40 @@ Author: Karel Piwko <kpiwko@redhat.com>
 ----------------------------
 
-This quickstart provides Arquillian tests. See [Run the Arquillian Tests](../README.html/#arquilliantests) for more information on how to set up and run the tests. When asked to start the JBoss Server, this quickstart requires that you start it with the _web_ profile.
+This quickstart provides Arquillian tests. See [Run the Arquillian Tests](../README.html/#arquilliantests) for information on how to set up and run the tests. When asked to start the JBoss Server, this quickstart requires that you start it with the _web_ profile.
```
My only concern with completely removing the info to another file is it requires users to click or navigate, which for a lazy user, may be too much.

sgilda replied:
I agree. But the alternative is to copy and paste everything into each file and maintain it everywhere. Should I do that? And if so, should all the sections be copied back into the individual quickstart READMEs (start the server, build and deploy, etc.)?

pmuir replied:
I'm not sure, let's chat on IRC.

sgilda replied:
I can certainly copy and paste the instructions back into the individual 
quickstart README files. I'm with you; there's a balance between 
navigating and maintaining the same text in multiple files.

On 03/29/2012 07:17 AM, Pete Muir wrote:

> > ---
> > 
> > -This quickstart provides Arquillian tests. See [Run the Arquillian Tests](../README.html/#arquilliantests) for more information on how to set up and run the tests. When asked to start the JBoss Server, this quickstart requires that you start it with the _web_ profile.
> > +This quickstart provides Arquillian tests. See [Run the Arquillian Tests](../README.html/#arquilliantests) for information on how to set up and run the tests. When asked to start the JBoss Server, this quickstart requires that you start it with the _web_ profile.
> > I'm not sure, let's chat on IRC.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167/files#r620498

pmuir commented on bmt/README.md:
```diff
@@ -1,96 +1,74 @@
+
+JBoss Enterprise Application Platform 6 and JBoss AS 7 ship with an H2 datasource by default. This example shows how to transactionally insert key value pairs into the H2 database and demonstrates the requirements on the developer with respect to the JPA Entity Manager.
+
+_NOTE: A JEE container is designed with robustness in mind, so you should carefully analyse the scaleabiltiy, concurrency and performance needs of you application before taking advantage of these techniques in your own applications._
```
Java EE, not JEE :-)

pmuir replied:
This one still needs fixing.

pmuir commented on ejb-in-war/README.md:
```diff
@@ -5,82 +5,62 @@ Author: Paul Robinson
+2. On clicking submit, the name is sent to a managed bean namde `Greeter`.
+3. On setting the name, the `Greeter` invokes the `GreeterEJB`, which was injected to the managed bean. Notice the field annotated with @EJB.
+4. The response from invoking the `GreeterEJB` is stored in a field (message) of the managed bean.
+5. The managed bean is annotated as @SessionScoped, so the same managed bean instance is used for the entire session. This ensures that the message is available when the page reloads and is displayed to the user.
```
@SessionScoped should have backticks

pmuir commented on ejb-remote/README.md:
```diff
@@ -5,65 +5,91 @@ Authors: Jaikiran Pai and Mike Musgrove
+1. A server side component:
+
+    The server component is comprised of a stateful and a stateless EJB. It provides both an EJB JAR that is deployed to the server and a JAR file containing the remote business interfaces required by the remote client application.
+2. A remote client program that accesses the server component. 
```
Application is better than program.

pmuir commented on ejb-remote/README.md:
```diff
@@ -5,65 +5,91 @@ Authors: Jaikiran Pai and Mike Musgrove
+Build and Deploy the Quickstart
+-------------------------
+
+Follow these steps to build, deploy and run the quickstart.
```
Worth explicitly noting that this differs from the root readme?

sgilda replied:
I should do that for the jts quickstart also.

sgilda commented:
Made changes suggested above. Let me know if I missed anything.

sgilda commented:
Added instructions to start the server to the individual quickstart READMEs and squashed the commits.

Should I copy in the build and deploy and arquillian instructions also?

pmuir commented:
Yes, I think so. Maybe just the mvn clean package jboss-as:deploy build instructions and arquillian remote tests. IOW we only show the basic steps in each readme…

On 29 Mar 2012, at 15:48, sgilda wrote:

> Added instructions to start the server to the individual quickstart READMEs and squashed the commits.
> 
> Should I copy in the build and deploy and arquillian instructions also?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167#issuecomment-4821691

sgilda commented:
I think we have to include the instructions for appending the Maven 
settings. Otherwise, it will fail for EAP 6 if it's not configured. What 
do you think?

On 03/29/2012 04:40 PM, Pete Muir wrote:

> Yes, I think so. Maybe just the mvn clean package jboss-as:deploy build instructions and arquillian remote tests. IOW we only show the basic steps in each readme…
> 
> On 29 Mar 2012, at 15:48, sgilda wrote:
> 
> > Added instructions to start the server to the individual quickstart READMEs and squashed the commits.
> > 
> > Should I copy in the build and deploy and arquillian instructions also?
> > 
> > ---
> > 
> > Reply to this email directly or view it on GitHub:
> > 
> > ## https://github.com/jbossas/quickstart/pull/167#issuecomment-4821691
> > 
> > Reply to this email directly or view it on GitHub:
> > https://github.com/jbossas/quickstart/pull/167#issuecomment-4830613

pmuir commented:
Maybe just the line "Make sure you have the EAP maven repository configured. For more, see …" and then omit it.

On 29 Mar 2012, at 21:56, sgilda wrote:

> I think we have to include the instructions for appending the Maven 
> settings. Otherwise, it will fail for EAP 6 if it's not configured. What 
> do you think?
> 
> On 03/29/2012 04:40 PM, Pete Muir wrote:
> 
> > Yes, I think so. Maybe just the mvn clean package jboss-as:deploy build instructions and arquillian remote tests. IOW we only show the basic steps in each readme…
> > 
> > On 29 Mar 2012, at 15:48, sgilda wrote:
> > 
> > > Added instructions to start the server to the individual quickstart READMEs and squashed the commits.
> > > 
> > > Should I copy in the build and deploy and arquillian instructions also?
> > > 
> > > ---
> > > 
> > > Reply to this email directly or view it on GitHub:
> > > 
> > > ## https://github.com/jbossas/quickstart/pull/167#issuecomment-4821691
> > > 
> > > Reply to this email directly or view it on GitHub:
> > > https://github.com/jbossas/quickstart/pull/167#issuecomment-4830613
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167#issuecomment-4830962

sgilda commented:
Added build and deploy. Added arquillian. Cleaned up some other issues I found.

sgilda commented:
I think this is ready for review again. Commits were quashed.

pmuir commented on README.md:
```diff
@@ -273,16 +281,20 @@ You can run these tests using either a managed or remote container.
+<a id="useeclipse"/>
+Use JBoss Developer Studio or Eclipse to Run the Quickstarts
+-------------------------------------
+You can also deploy the quickstarts from Eclipse using JBoss tools. For more information on how to set up Maven and the JBoss tools, refer to the [Boss Enterprise Application Platform 6 Beta](http://docs.redhat.com/docs/en-US/JBoss_Enterprise_Application_Platform/6/html/Beta_Documentation/HOME.html) documentation or the [Getting Started Developing Applications Guide](https://docs.jboss.org/author ...
```
Should be JBoss not Boss :-)

sgilda replied:
Done. :-)

pmuir commented on README.md:
```diff
@@ -201,42 +201,46 @@ The README for each quickstart will specify which configuration is required to r
-            mvn jboss-as:undeploy
-   6. If you want to redeploy a quickstart that uses a database, you should undeploy it first. This ensures the database is cleaned up correctly before the redeploy.
+            mvn clean package jboss-as:deploy
+
```
Probably I swap the order of this  - Build & Deploy and then Build Only.

sgilda replied:
The reason I put it in this order was because during the development 
phase, you often don't even start your server. You only build in order 
to clear up compile and build errors. It's not until you get to the 
testing phase that you actually deploy. But whatever you think is fine 
by me.

On 04/01/2012 06:16 AM, Pete Muir wrote:

> > -            mvn jboss-as:undeploy
> > -   6. If you want to redeploy a quickstart that uses a database, you should undeploy it first. This ensures the database is cleaned up correctly before the redeploy.
> > -            mvn clean package jboss-as:deploy
> >   +
> >   Probably I swap the order of this  - Build&  Deploy and then Build Only.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167/files#r630609

sgilda replied:
Then should the "Build the Archive" Build be after the Undeploy as well? It looks strange sandwiched between "Build and Deploy the Archive" and "Undeploy the Archive".
I still think I like it first since it explains that it only builds to test for compile errors and the next logical step is to build and deploy.

pmuir replied:
Yep, my thought was that for the quickstarts we've already done the cleaning up of compile and build errors. First thing people want to do is deploy…

On 1 Apr 2012, at 15:18, sgilda wrote:

> > -            mvn jboss-as:undeploy
> > -   6. If you want to redeploy a quickstart that uses a database, you should undeploy it first. This ensures the database is cleaned up correctly before the redeploy.
> > -            mvn clean package jboss-as:deploy
> >   +
> 
> The reason I put it in this order was because during the development 
> phase, you often don't even start your server. You only build in order 
> to clear up compile and build errors. It's not until you get to the 
> testing phase that you actually deploy. But whatever you think is fine 
> by me.
> 
> On 04/01/2012 06:16 AM, Pete Muir wrote:
> 
> > > -            mvn jboss-as:undeploy
> > > -   6. If you want to redeploy a quickstart that uses a database, you should undeploy it first. This ensures the database is cleaned up correctly before the redeploy.
> > > -            mvn clean package jboss-as:deploy
> > >   +
> > >   Probably I swap the order of this  - Build&  Deploy and then Build Only.
> > 
> > ---
> > 
> > Reply to this email directly or view it on GitHub:
> > https://github.com/jbossas/quickstart/pull/167/files#r630609
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167/files#r630733

pmuir commented on README.md:
```diff
@@ -326,7 +338,7 @@ The following procedures describe how to add a user with the appropriate permiss
         Realm (ManagementRealm) : 
 
-    At the prompt, press enter to take the default ManagementRealm
+    If the quickstart README specifies a realm, type it here. Otherwise, press enter to take the default `ManagementRealm`. 
```
maybe s/take/use/ ?

sgilda replied:
Done.

pmuir commented on bean-validation/README.md:
```diff
@@ -45,5 +67,20 @@ Search for lines similar to the following ones in the server output log:
+
+Run the Quickstart in JBoss Developer Studio or Eclipse
+-------------------------------------
+You can also start the server and deploy the quickstarts from Eclipse using JBoss tools. For more information, see [Use JBoss Developer Studio or Eclipse to Run the Quickstarts](../README.html/#useeclipse) 
```
Probably should be "test" not "run", as this is not a deployable quickstart.

sgilda replied:
Done.

pmuir commented on cdi-injection/README.md:
```diff
@@ -5,49 +5,72 @@ Author: Jason Porter
 
-This example demonstrates the use of *CDI 1.0 Injection and Qualifiers* in *JBoss AS 7*  or *JBoss Enterprise Application Platform 6* 
-with JSF as the front-end client.
+This example demonstrates the use of *CDI 1.0 Injection and Qualifiers* in *JBoss AS 7*  or *JBoss Enterprise Application Platform 6* with JSF as the front-end client.
```
Swap JBoss AS 7 and EAP.

sgilda replied:
Done.

pmuir commented on cmt/README.md:
```diff
@@ -17,104 +14,102 @@ Aspects touched upon in the code:
-this stateless session bean has been marked up with an @javax.ejb.TransactionAttribute annotation.
+### What makes this an example of container managed transactions?
+
+Take a look at <code>org.jboss.as.quickstarts.cmt.ejb.CustomerManagerEJBImpl</code>. You can see that this stateless session bean has been marked up with an @javax.ejb.TransactionAttribute annotation.
```
s/an/the/

sgilda replied:
I assume you meant the 'an' in this line, not the title? 
You can see that this stateless session bean has been marked up with the @javax.ejb.TransactionAttribute annotation.

Done.

pmuir commented on cmt/README.md:
```diff
@@ -17,104 +14,102 @@ Aspects touched upon in the code:
+* Never - If the client has a transaction running and does not suspend it but calls a method annotated with Never then an EJB exception will be raised.
+
+
+JBoss Enterprise Application Platform 6 and JBoss AS 7 ship with an H2 datasource by default. This example shows how to transactionally insert key value pairs into the H2 database and demonstrates the requirements on the developer with respect to the JPA Entity Manager.
```
This is no longer right, as we deploy our own datasource. Perhaps it should say "ship with H2, an in memory database written in Java.".

sgilda replied:
Sounds good. I believe we need a dash for 'in-memory' though. If you google it, almost every site uses the dash, including Wikipedia, IBM, and Oracle.

pmuir replied:
Sounds good.

On 1 Apr 2012, at 15:33, sgilda wrote:

> > -\* Supports - This will run the method within a transaction if a transaction exists, 
> > -alternatively, if there is no transaction running the method will not be executed 
> > -within the scope of a transaction 
> > -\* Never - If the client has a transaction running and does not suspend it but calls 
> > 
> > ## -a method annotated with Never then an EJB exception will be raised.
> > 
> > -## System requirements
> > +\* Required - As demonstrated in the example. If a transaction does not already exist, this will initiate a transaction and  complete it for you, otherwise the business logic will be integrated into the existing transaction.
> > +\* RequiresNew - If there is already a transaction running, it will be suspended, the work performed within a new transaction which is completed at exit of the method and then the original transaction resumed. 
> > +\* Mandatory - If there is no transaction running, calling a business method with is annotated   with this will result in an error.
> > +\* NotSupported - If there is a transaction running, it will be suspended and no transaction will be initiated for this business method.
> > +\* Supports - This will run the method within a transaction if a transaction exists, alternatively, if there is no transaction running the method will not be executed within the scope of a transaction. 
> > +\* Never - If the client has a transaction running and does not suspend it but calls a method annotated with Never then an EJB exception will be raised.
> > +
> > +
> > +JBoss Enterprise Application Platform 6 and JBoss AS 7 ship with an H2 datasource by default. This example shows how to transactionally insert key value pairs into the H2 database and demonstrates the requirements on the developer with respect to the JPA Entity Manager.
> 
> Sounds good. I believe we need a dash for 'in-memory' though. If you google it, almost every site uses the dash, including Wikipedia, IBM, and Oracle.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167/files#r630747

pmuir commented on README.md:
```diff
@@ -170,7 +170,7 @@ The JBoss server can be started a few different ways.
 
 <a id="startserverweb"></a>
-#### Start the JBoss Enterprise Application Platform 6 or JBoss AS 7 Server with the Web Profile
+#### Start JBoss Enterprise Application Platform 6 or JBoss Application Server 7 with the Web Profile
```
We don't want to expand out JBoss AS 7 I think - JBoss AS 7 is its brand, just like JBoss Enterprise Application Platform is its.

sgilda replied:
I saw it expanded in the branding image, so I assumed it had to be spelled out.  I've changed everything back to JBoss AS 7..

pmuir replied:
Lol!

I guess we should check this with James.

pmuir replied:
http://www.jboss.org/as7

On 1 Apr 2012, at 15:52, sgilda wrote:

> > @@ -170,7 +170,7 @@ The JBoss server can be started a few different ways.
> > The README for each quickstart will specify which configuration is required to run the example.
> > 
> > <a id="startserverweb"></a>
> > -#### Start the JBoss Enterprise Application Platform 6 or JBoss AS 7 Server with the Web Profile
> > +#### Start JBoss Enterprise Application Platform 6 or JBoss Application Server 7 with the Web Profile
> 
> I saw it expanded in the branding image, so I assumed it had to be spelled out.  I've changed everything back to JBoss AS 7..
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167/files#r630773

pmuir commented on cmt/README.md:
```diff
@@ -17,104 +14,102 @@ Aspects touched upon in the code:
+
+You will be presented with a simple form for adding customers to a database.
+
+After a user is successfully added to the database, a message is produced container the details of the user. An example MDB will dequeue this message and print the contents as such:
```
I would remove "as such", it's a somewhat odd expression ;-)

sgilda replied:
I didn't edit much of what was there. I agree about "as such". Changed it to read: "An example MDB will dequeue this message and print the following contents:"

This also doesn't make any sense: "a message is produced container the details of the user"
I believe they meant: "a message is produced containing the details of the user"

pmuir replied:
Yes, I know I'm commenting on stuff you didn't change, but I think it's worth fixing up whilst we are here :-)

On 1 Apr 2012, at 15:57, sgilda wrote:

> > -After a user is successfully added to the database, a message is produced container the 
> > -details of the user. An example MDB will dequeue this message and print the contents as such:
> > +
> > +Access the application 
> > +---------------------
> > +
> > +The application will be running at the following URL:  http://localhost:8080/jboss-as-cmt/.
> > +
> > +You will be presented with a simple form for adding customers to a database.
> > +
> > +After a user is successfully added to the database, a message is produced container the details of the user. An example MDB will dequeue this message and print the contents as such:
> 
> I didn't edit much of what was there. I agree about "as such". Changed it to read: "An example MDB will dequeue this message and print the following contents:"
> 
> This also doesn't make any sense: "a message is produced container the details of the user"
> I believe they meant: "a message is produced containing the details of the user"
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167/files#r630782

pmuir commented on ejb-in-ear/README.md:
```diff
@@ -41,39 +40,40 @@ Configure Maven
+2. Open a command line and navigate to the root directory of this quickstart.
+3. Type this command to undeploy the archive:
 
+            mvn jboss-as:undeploy
```
I think this is indented by 3 too many spaces.

sgilda replied:
This indent is the same as the one above for 'mvn clean package...". 
Changed both.

pmuir commented on ejb-in-war/README.md:
```diff
@@ -5,82 +5,82 @@ Author: Paul Robinson
-5. The managed bean is annotated as @SessionScoped, so the same managed bean instance is used for the entire session. This ensures that the message is available when the page reloads and is
-displayed to the user.
+2. On clicking submit, the name is sent to a managed bean namde `Greeter`.
+3. On setting the name, the `Greeter` invokes the `GreeterEJB`, which was injected to the managed bean. Notice the field annotated with @EJB.
```
s/to the/into the/, backticks for `@EJB`

sgilda replied:
Done

pmuir commented on ejb-in-war/README.md:
```diff
@@ -5,82 +5,82 @@ Author: Paul Robinson
-displayed to the user.
+2. On clicking submit, the name is sent to a managed bean namde `Greeter`.
+3. On setting the name, the `Greeter` invokes the `GreeterEJB`, which was injected to the managed bean. Notice the field annotated with @EJB.
+4. The response from invoking the `GreeterEJB` is stored in a field (message) of the managed bean.
```
backticks for `message`, and remove the brackets.

pmuir commented on README.md:
```diff
@@ -170,7 +170,7 @@ The JBoss server can be started a few different ways.
-#### Start the JBoss Enterprise Application Platform 6 or JBoss AS 7 Server with the Web Profile
+#### Start JBoss Enterprise Application Platform 6 or JBoss Application Server 7 with the Web Profile
 
 1. Open a command line and navigate to the root of the JBoss directory.
```
Sorry, should have picked this up before - needs to JBoss Server.

pmuir replied:
Maybe even better is to generally use "JBoss server" - JBoss Server isn't a thing, we're just describing a "server from JBoss".

sgilda replied:
Are you saying I should replace : 
Start the JBoss Enterprise Application Platform 6 or JBoss AS 7 Server

wih:
Start the JBoss Server
??

pmuir replied:
Nope. I'm saying "root of the JBoss directory" -> "root of the JBoss server directory". JBoss as a brand is overloaded, and we have to be careful using it.

sgilda replied:
This was why I liked linking to the instructions in one file. :-)

pmuir commented on ejb-remote/README.md:
```diff
@@ -5,65 +5,94 @@ Authors: Jaikiran Pai and Mike Musgrove
-by the remote client application.
+1. A server side component:
+
+    The server component is comprised of a stateful and a stateless EJB. It provides both an EJB JAR that is deployed to the server and a JAR file containing the remote business interfaces required by the remote client application.
```
I would say "stateful EJB and stateless EJB", it's a bit more readable.

pmuir commented on README.md:
```diff
@@ -179,7 +179,7 @@ The README for each quickstart will specify which configuration is required to r
 
 <a id="startserverfull"></a>
-#### Start the JBoss Enterprise Application Platform 6 or JBoss AS 7 Server with the Full Profile
+#### Start JBoss Enterprise Application Platform 6 or JBoss Application Server 7 with the Full Profile
```
Here too, just JBoss AS 7

pmuir commented on ejb-security/README.md:
```diff
@@ -2,66 +2,95 @@ ejb-security:  Using JEE Declarative Security to Control Access to EJB 3
+What is it?
+-----------
+
+This example demonstrates the use of JEE declarative security to control access to EJB 3 and Security in *JBoss Enterprise Application Platform 6* or *JBoss AS 7*.
```
change JEE to Java EE, EJB 3 to EJB3

pmuir commented on ejb-security/README.md:
```diff
@@ -2,66 +2,95 @@ ejb-security:  Using JEE Declarative Security to Control Access to EJB 3
-4. A security constraint is added to `WEB-INF/web.xml` 
-5. Security annotations are added to the EJB declaration
+1. Use the security domain `other` that is defined in the application server default `standalone.xml` configuration file.
+2. Add the user `quickstartUser` with password `quickstartPassword`, belonging to the role `guest` and realm `ApplicationRealm`. The procedure for adding a user is described below.
```
Actually, they are described in the root readme.

sgilda replied:
I think he was trying to give an outline of what will be done in this quickstart. It does say "The procedure for adding a user is described below." 
Should I remove that from the beginning since it's added later in the quickstart?
I think this one needs more work.

pmuir commented on ejb-security/README.md:
```diff
@@ -2,66 +2,95 @@ ejb-security:  Using JEE Declarative Security to Control Access to EJB 3
+Add an Application User
+---------------
+
+This quickstart uses a secured management interface and requires that you create an application user to access the running application. Instructions to set up an Application user can be found here:  [Add an Application User](../README.html/#addapplicationuser)
```
This might be a bit confusing, as it's covered above in "What is it".

sgilda replied:
I would prefer to leave this section and remove it from the "What is it?"
It is more likely to be overlooked in the beginning section.

pmuir replied:
Sounds good. Right now it doesn't flow well :-)

On 1 Apr 2012, at 16:12, sgilda wrote:

> > +
> > +All you need to build this project is Java 6.0 (Java SDK 1.6) or better, Maven 3.0 or better.
> > +
> > +The application this project produces is designed to be run on JBoss Enterprise Application Platform 6 or JBoss AS 7. 
> > +
> > +
> > +Configure Maven
> > +---------------
> > +
> > +If you have not yet done so, you must [Configure Maven](../README.html/#mavenconfiguration) before testing the quickstarts.
> > +
> > +
> > +Add an Application User
> > +---------------
> > +
> > +This quickstart uses a secured management interface and requires that you create an application user to access the running application. Instructions to set up an Application user can be found here:  [Add an Application User](../README.html/#addapplicationuser)
> 
> I would prefer to leave this section and remove it from the "What is it?"
> It is more likely to be overlooked in the beginning section.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167/files#r630801

sgilda replied:
I agree. I think the overview needs to be removed or made clear that it's an overview. How about this:

This quickstart takes the following steps to implement EJB security:
1. Define a security domain in the `standalone.xml` configuration file.
2. Add an application user with access rights to the application.
3. Add a security domain reference to `WEB-INF/jboss-web.xml`.
4. Add a security constraint to the `WEB-INF/web.xml` .
5. Add a security annotation to the EJB declaration.

pmuir replied:
Looks good.

On 1 April 2012 16:47, sgilda
reply@reply.github.com
wrote:

> > +
> > +All you need to build this project is Java 6.0 (Java SDK 1.6) or better, Maven 3.0 or better.
> > +
> > +The application this project produces is designed to be run on JBoss Enterprise Application Platform 6 or JBoss AS 7.
> > +
> > +
> > +Configure Maven
> > +---------------
> > +
> > +If you have not yet done so, you must [Configure Maven](../README.html/#mavenconfiguration) before testing the quickstarts.
> > +
> > +
> > +Add an Application User
> > +---------------
> > +
> > +This quickstart uses a secured management interface and requires that you create an application user to access the running application. Instructions to set up an Application user can be found here:  [Add an Application User](../README.html/#addapplicationuser)
> 
> I agree. I think the overview needs to be removed or made clear that it's an overview. How about this:
> 
> This quickstart takes the following steps to implement EJB security:
> 1. Define a security domain in the `standalone.xml` configuration file.
> 2. Add an application user with access rights to the application.
> 3. Add a security domain reference to `WEB-INF/jboss-web.xml`.
> 4. Add a security constraint to the `WEB-INF/web.xml` .
> 5. Add a security annotation to the EJB declaration.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167/files#r630827

pmuir commented on h2-console/README.md:
```diff
@@ -5,83 +5,74 @@ Author: Pete Muir
+Additional requirements
+-----------------------
+
+You will need the H2 Console to run test this quickstart. You can download it from <http://www.h2database.com/html/download.html>. We recommend using the platform independent zip.
```
No you don't. It's bundled in the quickstart directory.

sgilda replied:
This has been there from the beginning. The diff is just offset. 
Removed.

pmuir commented:
I think it could be a good idea to put at the top of those quickstart's readmes that are covered in the Getting Started Developing Applications Guide, that they are covered there...

pmuir commented on helloworld-rs/README.md:
```diff
@@ -7,31 +7,88 @@ What is it?
 
-Deploying the application
+ 
+Build and Deploy the Quickstart - Locally
```
I'm not sure I like the Locally. I would prefer that this is the "default case", and then we use OpenShift as the other case.

sgilda replied:
The Aerogear HTML5 quickstarts use this term too. Should they change it there too?

pmuir commented on helloworld-rs/README.md:
```diff
@@ -7,31 +7,88 @@ What is it?
+3. Type this command to undeploy the archive:
 
-    mvn package jboss-as:deploy 
+            mvn jboss-as:undeploy
```
This looks like too  many spaces for indent - should be 8.

pmuir commented on helloworld-rs/README.md:
```diff
@@ -7,31 +7,88 @@ What is it?
+2. Open a command line and navigate to the root directory of this quickstart.
+3. Type this command to build and deploy the archive:
+
+            mvn clean package jboss-as:deploy
```
This looks like too  many spaces for indent - should be 8.

pmuir commented on hibernate4/README.md:
```diff
@@ -5,34 +5,38 @@ Author: Madhumita Sadhukhan
 
-JBoss AS7 (or EAP 6) provides both JPA as well as Hibernate4(as ORM) and also supports Hibernate3 etc.
-If you choose to use Hibernate4 packaged within JBoss AS7(or EAP6) you will need to first import the JPA API.
+JBoss Enterprise Application Platform 6 and JBoss AS 7 both provide Hibernate 3, Hibernate 4 (as ORM), and JPA support. 
```
I think "(as ORM)" doesn't really add anything to this sentence...

pmuir commented on hibernate4/README.md:
```diff
@@ -47,66 +51,61 @@ for eg .
       </dependency>
 
-You may refer "Adding a new external dependency" located at http://community.jboss.org/wiki/HackingOnAS7 for further help on adding dependencies.
+You may refer "Adding a new external dependency" located at <http://community.jboss.org/wiki/HackingOnAS7> for further help on adding dependencies.
```
This is a really bad reference to point people to, as it's targeted at JBoss AS developers. We need to find a better one.

pmuir replied:
Probably just remove it.

pmuir commented on jax-rs-client/README.md:
```diff
@@ -1,39 +1,63 @@
+-----------
+
+This example demonstrates an external JAX-RS RestEasy client which interacts with a JAX-RS Web service that uses *CDI 1.0* and *JAX-RS* 
+in *JBoss Enterprise Application Platform 6* or *JBoss AS 7*.  
```
Erroneous new line?

sgilda replied:
Are you saying I should not put "This client calls..." in a new paragraph?

Most of the quickstarts have one paragraph with the technologies demonstrated and a new paragraph for description of the quickstart.

pmuir commented on kitchensink-ear/README.md:
```diff
@@ -5,9 +5,9 @@ Author: Pete Muir
 -----------
 
-This is your project! It's a sample, deployable Maven 3 project to help you get your foot in the door developing with Java EE 6 on JBoss AS 7 or JBoss Enterprise Application Platform 6. 
+This is your project! It is a sample, deployable Maven 3 project to help you get your foot in the door developing with Java EE 6 on JBoss Enterprise Application Platform 6 or JBoss AS 7. 
```
Can you send pulls for this stuff to the archetypes as well, so we keep them in sync?

sgilda replied:
I added an issue so we remember when it's complete

pmuir commented on kitchensink-jsp/README.md:
```diff
@@ -5,97 +5,91 @@ Author: Elvadas Nono
-Member Registration template.
-It reused the persistence unit and some sample persistence and transaction code to help 
-you get your feet wet with database access in enterprise Java. 
+This project rebuilds the presentation tier of the `kitchensink` quickstart using JSP and JSTL instead of JSF features. It reuses all other components from the Member Registration template. It also reuses the persistence unit and some sample persistence and transaction code to help you with database access in enterprise Java. 
```
rebuilds is perhaps overloaded, recreates might be better?

pmuir commented on kitchensink/README.md:
```diff
@@ -1,86 +1,85 @@
-This is your project! It's a sample, deployable Maven 3 project to help you get your foot in the door developing with Java EE 6 on JBoss Enterprise Application Platform 6 or JBoss AS 7. This project is setup to allow you to create a compliant Java EE 6 application using JSF 2.0, CDI 1.0, EJB 3.1, JPA 2.0 and Bean Validation 1.0. It includes a persistence unit and some sample persistence and trans ...
+This is your project! It is a sample, deployable Maven 3 project to help you get your foot in the door developing with Java EE 6 on JBoss Enterprise Application Platform 6 or JBoss AS 7. 
+
+This project is setup to allow you to create a compliant Java EE 6 application using JSF 2.0, CDI 1.0, EJB 3.1, JPA 2.0 and Bean Validation 1.0. It includes a persistence unit and some sample persistence and transaction code to introduce you to database access in enterprise Java. 
```
Likewise, can you send these changes the way of the archetypes as well?

pmuir commented on log4j/README.md:
```diff
@@ -5,77 +5,78 @@ Author: Bartosz Baranowski
-<a href="https://docs.jboss.org/author/display/AS7/Developer+Guide#DeveloperGuide-JBossDeploymentStructureFile"title="JBoss Deployment Structure File">JBoss Deployment Structure File</a>  
+Applications must often depend on third-party libraries. By default, Java EE packages allow you to include dependencies in a deployable unit which can lead to uncontrolled growth of the deployable unit. This can be avoided by the use of container defined modules. A module is nothing more than a container managed binary dependency which is shared by all deployed applications. For more informatin o ...
 
+This example is very simple. It declares dependency on Apache Log4j module which allows it to use a custom logging framework. This is achieved with a simple addition to the xml file: **src/main/webapp/WEB-INF/jboss-deployment-structure.xml**. For more information about this file please refer to [JBoss Deployment Structure File](https://docs.jboss.org/author/display/AS7/Developer+Guide#DeveloperGu ...
```
I would suggest we just make these simple (no description) links - I don't think that adds much...

sgilda replied:
I'm unclear about the change you are requesting here.

pmuir replied:
"[JBoss Deployment Structure File](https://docs.jboss.org/author/display/AS7/Developer+Guide#DeveloperGuide-JBossDeploymentStructureFile)." -> just a plain angle brackets link.

pmuir commented on mail/README.md:
```diff
@@ -7,8 +7,9 @@ What is it?
-The configuration of the mail provider is found in either *$JBOSS_HOME/standalone/configuration.xml* or *$JBOSS_HOME/domain/configuration.xml*. An example is provided below:
+The example uses the default Mail provider that comes out of the box with *JBoss Enterprise Application Platform 6* and *JBoss AS 7*.  It uses your local mail relay and the default SMTP port of 25.
+
+The configuration of the mail provider is found in either *JBOSS_HOME/standalone/configuration.xml* or *JBOSS_HOME/domain/configuration.xml*. An example is provided below:
```
These stars should be backticks

pmuir commented on servlet-security/README.md:
```diff
@@ -2,66 +2,96 @@ servlet-security:  Using JEE Declarative Security to Control Access to Servlet 3
+What is it?
+-----------
+
+This example demonstrates the use of JEE declarative security to control access to Servlets and Security in JBoss Enterprise Application Platform 6 and  JBoss AS7.
```
Java EE not JEE.

pmuir commented on servlet-security/README.md:
```diff
@@ -2,66 +2,96 @@ servlet-security:  Using JEE Declarative Security to Control Access to Servlet 3
 
 1. Use the security domain `other` that is defined in the application server's default `standalone.xml`
-2. Add the user `user` with password `password` belonging to the role `guest` and realm `ApplicationRealm`. For more information regarding how to add a user using the "add-user" utility, please refer to the README in the root of the quickstarts 
+2. Add the user `quickstartUser` with password `quickstartPassword` belonging to the role `guest` and realm `ApplicationRealm`. The procedure for adding a user is described below.
```
As before, the flow isn't quite right here.

pmuir commented on wsat-simple/README.md:
```diff
@@ -19,15 +19,14 @@ For a more complete example, please see the XTS demonstrator application that sh
 2. An operation on a WS-AT enabled Web service is invoked by the client.
 3. The JaxWSHeaderContextProcessor in the WS Client handler chain inserts the WS-AT context into the outgoing SOAP message
-4. When the service receives the SOAP request, it's JaxWSHeaderContextProcessor in it's handler chain inspects the WS-AT context and associates the request with this AT.
+4. When the service receives the SOAP request, the JaxWSHeaderContextProcessor in its handler chain inspects the WS-AT context and associates the request with this AT.
```
the JaxWS... should be in backticks

pmuir commented on wsat-simple/README.md:
```diff
@@ -19,15 +19,14 @@ For a more complete example, please see the XTS demonstrator application that sh
 
-When running the org.jboss.as.quickstarts.wsat.simple.ClientTest#testCommit() method, the
-following steps occur:
+When running the org.jboss.as.quickstarts.wsat.simple.ClientTest#testCommit() method, the following steps occur:
```
Use backticks around the method.

pmuir commented on wsat-simple/README.md:
```diff
@@ -40,37 +39,52 @@ There is another test that shows what happens if the client decides to rollback
+3. Type the following command to run the test goal with the following profile activated:
 
-    mvn clean test -Parq-jbossas-remote
+                  mvn clean test -Parq-jbossas-remote 
```
Should be 8 spaces.

pmuir commented on wsba-coordinator-completion-simple/README.md:
```diff
@@ -20,57 +17,70 @@ quick start to demonstrate more advanced features. In particular:
 
-It is also assumed that you have an understanding of WS-BusinessActivity. For more details, read the XTS documentation
-that ships with the JBossTS project, which can be downloaded here: http://www.jboss.org/jbosstm/downloads/JBOSSTS_4_16_0_Final
+It is also assumed that you have an understanding of WS-BusinessActivity. For more details, read the XTS documentation that ships with the JBossTS project, which can be downloaded here: http://www.jboss.org/jbosstm/downloads/JBOSSTS_4_16_0_Final
```
Link brackets  needed :-)

pmuir commented on wsba-coordinator-completion-simple/README.md:
```diff
@@ -20,57 +17,70 @@ quick start to demonstrate more advanced features. In particular:
 
-When running the org.jboss.as.quickstarts.wsba.coordinatorcompletion.simple.ClientTest#testSuccess() method, the
-following steps occur:
+When running the org.jboss.as.quickstarts.wsba.coordinatorcompletion.simple.ClientTest#testSuccess() method, the following steps occur:
```
Backticks around method name

pmuir commented on wsba-coordinator-completion-simple/README.md:
```diff
@@ -20,57 +17,70 @@ quick start to demonstrate more advanced features. In particular:
-3. The JaxWSHeaderContextProcessor in the WS Client handler chain inserts the BA context into the outgoing SOAP messages
-4. When the service receives a SOAP request, it's JaxWSHeaderContextProcessor in it's handler chain inspects the BA context and associates the request with this BA.
-5. The Web service operation is invoked...
+3. The JaxWSHeaderContextProcessor in the WS Client handler chain inserts the BA context into the outgoing SOAP messages.
```
Backticks around JaxWS...

pmuir commented on wsba-coordinator-completion-simple/README.md:
```diff
@@ -20,57 +17,70 @@ quick start to demonstrate more advanced features. In particular:
 7. The service invokes the business logic. In this case, a String value is added to the set.
-9. The client can then make additional calls to the SetService. As the SetService participates as a CoordinatorCompletion protocol, it will continue to accept calls to 'addValueToSet' until it is told to complete by the coordinator.
-10. The client can then decide to complete or cancel the BA. If the client decides to complete, all participants will be told to complete. Providing all participants successfully complete, the coordinator will then tell all participants to close, otherwise the completed participants will be told to compensate.  If the participant decides to cancel, all participants will be told to compensate.
+9. The client can then make additional calls to the SetService. As the SetService participates as a CoordinatorCompletion protocol, it will continue to accept calls to `addValueToSet` until it is told to complete by the coordinator.
```
backticks around class names

pmuir commented on wsba-participant-completion-simple/README.md:
```diff
@@ -31,13 +28,13 @@ following steps occur:
-3. The JaxWSHeaderContextProcessor in the WS Client handler chain inserts the BA context into the outgoing SOAP message
-4. When the service receives the SOAP request, it's JaxWSHeaderContextProcessor in it's handler chain inspects the BA context and associates the request with this BA.
-5. The Web service operation is invoked...
+3. The JaxWSHeaderContextProcessor in the WS Client handler chain inserts the BA context into the outgoing SOAP message.
```
backticks around class names

pmuir commented:
Ok, I've been through all of it. I'll need to do a pass through the whole lot before we go FInal, but I think we are looking pretty good.

Sande, do you want to merge once you've applied changes?

sgilda commented:
How does this sound?

There is a tutorial for this quickstart in the [Getting Started 
Developing Applications 
Guide](https://docs.jboss.org/author/display/AS71/Getting+Started+Developing+Applications+Guide).

On 04/01/2012 07:19 AM, Pete Muir wrote:

> I think it could be a good idea to put at the top of those quickstart's readmes that are covered in the Getting Started Developing Applications Guide, that they are covered there...
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167#issuecomment-4865751

sgilda committed:
More readme cleanup

sgilda commented:
I think I got most of the changes. Like you said, we'll need another pass through them.
Merged and closed.

pmuir commented:
Sounds good

On 1 April 2012 16:53, sgilda
reply@reply.github.com
wrote:

> How does this sound?
> 
> There is a tutorial for this quickstart in the [Getting Started
> Developing Applications
> Guide](https://docs.jboss.org/author/display/AS71/Getting+Started+Developing+Applications+Guide).
> 
> On 04/01/2012 07:19 AM, Pete Muir wrote:
> 
> > I think it could be a good idea to put at the top of those quickstart's readmes that are covered in the Getting Started Developing Applications Guide, that they are covered there...
> > 
> > ---
> > 
> > Reply to this email directly or view it on GitHub:
> > https://github.com/jbossas/quickstart/pull/167#issuecomment-4865751
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jbossas/quickstart/pull/167#issuecomment-4867138
